packages/medusa/src/loaders/api.ts

```diff
@@ -35,14 +33,27 @@ export default async ({ app, configModule }: Options) => {
      */
     await new RoutesLoader({
       app: app,
+      configModule: container.resolve(ContainerRegistrationKeys.CONFIG_MODULE),
       rootDir: path.join(__dirname, "../api-v2"),
-      configModule,
     }).load()
   } catch (err) {
     throw Error(
-      "An error occurred while registering Medusa Core API Routes. See error in logs for more details."
+      "An error occurred while registering Medusa Core API Routes. See error in logs for more details.",
+      { cause: err }
     )
   }
 
+  await Promise.all(
```

Comment:
Question: I believe the route from the plugins was loaded first in v1, making it possible to override or prepend middlewares to an existing route from the core. Here, we will not be able to to that anymore. I believe it requires a bit more discussion but I wanted to mention it there, wdyt?

Reply:
No idea. I am not aware how things were done in v1 :)

But, if plugin is loaded after, wouldn't it be able to override middleware loaded before it?

Reply:
No i dont think so as it will be added after therefore in order to resolve a request express will go through the matchers and encounter the core middlewares and then the core route which returns and stop there, so the plugins wont be reached for this scenario

Reply:
Okay, I think I will have to pair with you to understand this :)